golem-worker-bridge/src/register.rs

```diff
@@ -140,29 +160,47 @@ impl RegisterApiDefinition for RedisApiRegistry {
         let key: ApiDefinitionKey = ApiDefinitionKey::from(definition);
         let redis_key = get_api_definition_redis_key(&key);
 
-        // if key already exists return conflict error
-        let value: Option<Bytes> = self
+        let value: u32 = self
             .pool
             .with("persistence", "get_definition")
-            .get(redis_key.clone())
+            .exists(redis_key.clone())
             .await
             .map_err(|e| ApiRegistrationError::InternalError(e.to_string()))?;
 
-        match value {
-            Some(_) => Err(ApiRegistrationError::AlreadyExists(key)),
-            None => {
-                let definition_value = self
-                    .pool
-                    .serialize(definition)
-                    .map_err(|e| ApiRegistrationError::InternalError(e.to_string()))?;
+        // if key already exists return conflict error
+        if value > 0 {
+            Err(ApiRegistrationError::AlreadyExists(key))
+        } else {
+            let definition_value = self
+                .pool
+                .serialize(definition)
+                .map_err(|e| ApiRegistrationError::InternalError(e.to_string()))?;
+
+            self.pool
+                .with("persistence", "register_definition")
+                .transaction({
+                    let api_id = key.id.clone();
+                    let api_version = key.version.0.clone();
+
+                    move |transaction| async move {
+                        let definition_key = get_api_definition_redis_key(&key);
+                        let version_set_key = get_api_definition_redis_key_version_set(&api_id);
+
+                        transaction
+                            .set(definition_key, definition_value, None, None, false)
+                            .await?;
+                        transaction.sadd(version_set_key, vec![api_version]).await?;
+                        transaction
+                            .sadd(get_all_apis_set_redis_key(), vec![api_id.0])
```

Comment:
it may be hard to say, how many api definitions ids we will have and how many versions for specific api definitions

if `get_all_apis_set_redis_key()` is mainly used for get_all which suppose to return all api definitions with all versions, probably it could be better if there will be `id:version` stored in this set

Reply:
The redis key for api definitions is actually `id:version` already in main branch

Reply:
I thought we will be able to fetch it with key prefix.

Reply:
it's possible to do with [SCAN](https://redis.io/commands/scan/), but it's not performant. I think that the two sets is overkill and it's possible to do with a single set `id:version` as peter suggested. I will fix this tomorrow